workers/tests/unittests/test_ohsome_client.py

```diff
@@ -40,15 +42,107 @@ def setUp(self) -> None:
 
     def test_query_valid_response(self) -> None:
         with patch("httpx.AsyncClient.post", new_callable=AsyncMock) as mock_request:
-            mock_request.return_value = httpx.Response(200, content=self.valid_geojson)
+            mock_request.return_value = httpx.Response(
+                200,
+                content=self.valid_geojson,
+                request=httpx.Request("POST", "url"),
+            )
             response = asyncio.run(ohsome_client.query(self.layer, self.bpolys))
             self.assertTrue(response.is_valid)
 
     def test_query_invalid_response_with_status_code_200(self) -> None:
         """When response is streamed it can be invalid while status code equals 200"""
         with patch("httpx.AsyncClient.post", new_callable=AsyncMock) as mock_request:
             mock_request.return_value = httpx.Response(
-                200, content=self.invalid_geojson
+                200,
+                content=self.invalid_geojson,
+                request=httpx.Request("POST", "url"),
             )
-            response = asyncio.run(ohsome_client.query(self.layer, self.bpolys))
-            self.assertIsNone(response)
+            with self.assertRaises(OhsomeApiError):
+                asyncio.run(ohsome_client.query(self.layer, self.bpolys))
+
+    def test_query_status_code_400(self) -> None:
+        with patch("httpx.AsyncClient.post", new_callable=AsyncMock) as mock_request:
+            mock_request.return_value = httpx.Response(
+                400,
+                content=self.invalid_geojson,
+                request=httpx.Request("POST", "url"),
+            )
+            with self.assertRaises(OhsomeApiError):
+                asyncio.run(ohsome_client.query(self.layer, self.bpolys))
+
+    def test_iso_time(self) -> None:
+        with self.assertRaises(ValueError):
+            ohsome_client.check_iso_time("")
+            ohsome_client.check_iso_time("20140101")
+
+        self.assertTrue(ohsome_client.check_iso_time("2014-01-01"))
+        self.assertTrue(ohsome_client.check_iso_time("2007-01-25T12:00:00"))
+
+        self.assertTrue(
+            ohsome_client.check_iso_time("2014-01-01,2015-07-01,2018-10-10")
+        )
+
+        # TODO:
+        # self.assertTrue(ohsome_client.check_iso_time("2007-01-25T12:00:00Z"))
+        # self.assertTrue(ohsome_client.check_iso_time("2014-01-01/2018-01-01/P1Y"))
```

Comment:
Before the PR is merged, I would suggest to either remove these two lines or make sure the either fail (`assertRaises`) or deliver the correct result.